### PR TITLE
Fixes #548 import of initial conditions does not overwrite existing

### LINKED ISSUE #548: import of molecular start values (from an Excel sheet) does not overwrite existing start values
Hello,
I need to open this issue now, I'll add details later!

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #548 import of initial conditions does not overwrite existing

# Description
This was because the trailing spaces were not removed from the imported path. Eg. `Organism|Liver||`

## Type of change

Please mark relevant options with an `x` in the brackets.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved import validation: correctly handle column counts, ignore rows without specified quantities, and trim trailing empty path segments for more reliable data imports.

* **Tests**
  * Added and updated test coverage for edge cases around path segment trimming and import-row handling.

### CODE REVIEW

File: src/MoBi.Presentation/Mappers/AbstractDataTableProviderToQuantityDTOMapper.cs [hunk 1]
```diff
@@ -65,14 +65,19 @@ protected IDimension GetDimension(DataTable table, int rowIndex, int unitColumn,
...
       /// <returns>The an enumeration of strings that can be used to create an ObjectPath</returns>
       protected static IEnumerable<string> GetPath(DataRow row, int index)
       {
-         return row[index].ToString().ToPathArray();
+         var path = row[index].ToString().ToPathArray().ToList();
+
+         while (path.Count > 0 && string.IsNullOrEmpty(path[path.Count - 1]))
+            path.RemoveAt(path.Count - 1);
+
+         return path;
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Mappers/AbstractDataTableProviderToQuantityDTOMapper.cs:80]: Changes.

I found some references to `msv` naming that I also corrected.

File: tests/MoBi.Tests/Presentation/Mapper/DataTableToImportQuantityDTOMapperForMoleculesSpecs.cs [hunk 2]
```diff
@@ -456,6 +456,31 @@ public void should_result_in_no_imported_quantities()
...
+      {
+         _result = sut.MapFrom(_tables, _initialConditionsBuildingBlock);
+      }
+
+      [Observation]
+      public void should_trim_trailing_empty_segments_from_the_path()
+      {
+         _result.QuantityDTOs[0].ContainerPath.Count.ShouldBeEqualTo(5);
+      }
+   }
```
Comment (rwmcintosh) [inline tests/MoBi.Tests/Presentation/Mapper/DataTableToImportQuantityDTOMapperForMoleculesSpecs.cs:482]: test for changes

File: src/MoBi.Presentation/Mappers/DataTableToImportQuantityDTOMapperForMolecules.cs [hunk 3]
```diff
@@ -90,14 +90,17 @@ private bool isCorrectDimensionForDimensionMode(ReactionDimensionMode dimensionM
...
       {
-         if (row.ItemArray.Count() < InitialConditions.COLUMNS)
+         if (row.ItemArray.Length < InitialConditions.COLUMNS)
             throw new ImportQuantityDTOsFromDataTablesMapperException(row, rowIndex, AppConstants.Exceptions.TableShouldBeNColumns(InitialConditions.COLUMNS));
 
-         var path = GetPath(row, InitialConditions.PATH);
+         var path = GetPath(row, InitialConditions.PATH).ToList();
 
+         while (path.Count > 0 && string.IsNullOrEmpty(path[path.Count - 1]))
+            path.RemoveAt(path.Count - 1);
```
[the remark this answers is not in the document]
Comment (rwmcintosh) [reply src/MoBi.Presentation/Mappers/DataTableToImportQuantityDTOMapperForMolecules.cs]: done

**Review by Yuri05 — APPROVED** [review]

**Comment by Yuri05** [comment]
> This was because the trailing spaces were not removed from the imported path. Eg. `Organism|Liver||`

why do we have something like `Organism|Liver||` in the exported paths?

**Comment by rwmcintosh** [comment]
> > This was because the trailing spaces were not removed from the imported path. Eg. `Organism|Liver||`
> 
> why do we have something like `Organism|Liver||` in the exported paths?

I don't see it when I export and the user does not specify how it happened. It actually shows up as this in the example he gave
image

So, not doubled, but just with a trailing pipe
